src/target/riscv/riscv-013.c

```diff
@@ -565,6 +589,11 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in,
 		bool *dmi_busy_encountered, int dmi_op, uint32_t address,
 		uint32_t data_out, int timeout_sec, bool exec, bool ensure_success)
 {
+	if (!target->dbgbase_set)
+		assert(target->dbgbase == 0);
```

Comment:
This can be re-written as:

```suggestion
	/* Safety check for dbgbase */
	assert(target->dbgbase_set || target->dbgbase == 0);
```